packaging/build_packages_goo.sh

```diff
@@ -26,11 +26,14 @@ trap exit_error ERR
 URL="http://metadata/computeMetadata/v1/instance/attributes"
 GCS_PATH=$(curl -f -H Metadata-Flavor:Google ${URL}/daisy-outs-path)
 BASE_REPO=$(curl -f -H Metadata-Flavor:Google ${URL}/base-repo)
+REPO=$(curl -f -H Metadata-Flavor:Google ${URL}/repo)
+PULL_REF=$(curl -f -H Metadata-Flavor:Google ${URL}/pull-ref)
 
 apt-get install -y git-core
-git clone "https://github.com/${BASE_REPO}/osconfig.git" 
-cd osconfig
-packaging/setup_goo.sh
+
+packaging/pull_repository.sh
```

Comment:
by putting the pull steps in a pull_repository script instead of a function, the variables set in *this* script will not be available to the pull_repository script. you either have to `export` all needed variables here, or use functions defined in and sourced from pull_repository

Reply:
exporting all the variables since they can be used in other subsequent scripts